### PR TITLE
[Enhancement] Optimize backup/restore judgment logic

### PR DESCRIPTION
**Pull request by Astralidea**
Fixes #27551

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/BackupRestoreAnalyzer.java [hunk 1]
```diff
@@ -308,19 +308,19 @@ public static void analyzeTableRef(TableRef tableRef, String dbName, Database db
...
                 throw new SemanticException("List partition table does not support backup/restore job");
             }
         }
 
-        if (olapTbl != null && partitionNames != null) {
+        if (partitionNames != null) {
             if (!tbl.isOlapOrCloudNativeTable()) {
                 ErrorReport.reportSemanticException(ErrorCode.ERR_WRONG_TABLE_NAME, tableName.getTbl());
             }
+            OlapTable olapTbl = (OlapTable) tbl;
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/BackupRestoreAnalyzer.java:323]: Does it work well when tbl is a cloud-native table?

Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/BackupRestoreAnalyzer.java:323]: Yes, cloud-native table is also olap table

**Review by srlch — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**